src/base.iced

```diff
@@ -100,13 +111,64 @@ exports.Base = class Base
   #-----------------------------------
 
   amount : () -> @icfg('amount', btcjs.networks.bitcoin.dustThreshold+1)
-  account : () -> @cfg('account') 
-  min_amount : () -> @amount() + btcjs.networks.bitcoin.feePerKb
-  max_amount : () -> 2*btcjs.networks.bitcoin.feePerKb
+  account : () -> @cfg('account')
+  fee_per_byte_limit : () -> @icfg('fee-per-byte-limit', 1000)
+  max_clearance_minutes : () -> @icfg('max-clearance-minutes', 240)
+  padding : () -> @fcfg('padding', 1)
   min_confirmations : () -> @icfg('min-confirmations', 3)
+  abs_min_marginal_fee: () -> btcjs.networks.bitcoin.feePerKb/1000
+  min_amount : () -> @amount() + @marginal_fee*1000
+
+  # Some reasonable lower bound on the total cost to transact
+  abs_min_amount : () -> @amount() + @abs_min_marginal_fee*1000
+
+  max_amount : () -> 2*@marginal_fee*1000
 
   #-----------------------------------
 
+  # Estimates fee per byte for a specific network type needed to achieve
+  # verification before maxClearanceMins minutes For bitcoin, returns in
+  # satoshis and uses the 21.co API with no fallback.
+  marginal_fee_estimator: ({type, maxClearanceMins}, cb) ->
+    esc = make_esc cb,'marginal_fee_estimator'
+    if type == 'bitcoin'
+      apiUrl = 'https://bitcoinfees.21.co/api/v1/fees/list'
+      await request apiUrl, esc defer resp, body
+      if resp.statusCode == 200
+        fees = JSON.parse(body)['fees']
```

Comment:
actually can you make this a little more robust?  `JSON.parse` can throw, and `fees` might be undefined. In either case the program will crash.